File: Cpp/Algorithm/src/Chapter3/RemoveDuplicates_26.cpp
#include "chapter3.h"
int myAlgo::RemoveDuplicates_26::removeDuplicates(vector<int>& nums)
{
	int i = 0;
	int size=nums.size();
	for (size_t j = 1; j < size; j++)
	{
		if (nums[i]!=nums[j])
		{
			nums[++i] = nums[j];
		}
	}
	return i + 1;
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/FourthRound/Chapter03/removeDuplicates_26.java
package algo.tzashinorpu.FourthRound.Chapter03;

public class removeDuplicates_26 {
    public int removeDuplicates(int[] nums) {
        // 0,0,1,1,1,2,2,3,3,4 -> 0,1,2,3,4,_,_,_,_,_
        // 1,1,2 ->
        int i = 0;
        for (int j = 1; j < nums.length; j++) {
            if (nums[i] != nums[j]) {
                nums[++i] = nums[j];
            }
        }
        return i + 1;
    }
}
